src/portfft/dispatcher/global_dispatcher.hpp

```diff
@@ -351,16 +356,32 @@ struct committed_descriptor<Scalar, Domain>::run_kernel_struct<Dir, LayoutIn, La
       for (Idx num_transpose = num_transposes - 1; num_transpose > 0; num_transpose--) {
         event = detail::transpose_level<Scalar, Domain>(
             kernels.at(static_cast<std::size_t>(num_transpose) + static_cast<std::size_t>(num_factors)),
-            static_cast<const Scalar*>(desc.scratch_ptr_1.get()), desc.scratch_ptr_2.get(), factors_and_scan,
-            committed_size, static_cast<Idx>(max_batches_in_l2), n_transforms, static_cast<IdxGlobal>(i), num_transpose,
-            num_factors, 0, desc.queue, desc.scratch_ptr_1, desc.scratch_ptr_2, {event});
+            desc.scratch_ptr_1.get(), desc.scratch_ptr_2.get(), factors_and_scan, committed_size,
+            static_cast<Idx>(max_batches_in_l2), n_transforms, static_cast<IdxGlobal>(i), num_factors, 0, desc.queue,
+            {event}, storage);
         event.wait();
+        if (storage == complex_storage::SPLIT_COMPLEX) {
+          event = detail::transpose_level<Scalar, Domain>(
+              kernels.at(static_cast<std::size_t>(num_transpose) + static_cast<std::size_t>(num_factors)),
+              desc.scratch_ptr_1.get() + imag_offset, desc.scratch_ptr_2.get() + imag_offset, factors_and_scan,
+              committed_size, static_cast<Idx>(max_batches_in_l2), n_transforms, static_cast<IdxGlobal>(i), num_factors,
+              0, desc.queue, {event}, storage);
+          event.wait();
+        }
+        desc.scratch_ptr_1.swap(desc.scratch_ptr_2);
```

Comment:
swapping of pointers needs to be in a `host_task`, which would depend on `event` returned from `transpose_level`. As of today, we do explicitly wait on that event, but that's something we would be fixing in the future, so it might be better if we pre-emptively move it inside a host task.

Reply:
Wrong. Remember, that was the bug we figured out and I thought would be enough to remove the waits?

Reply:
Yes, we thought that kernel was being enqued before the pointers got swapped, but that didn't work. Hence we still have that event.wait(). What I was suggesting is that since we will be removing that event.wait at some point, 
Let's move there swapping of the pointers by making it wait for that aforementioned event by moving it inside of a host task

Reply:
After testing again I figured out I can remove the waits between transpose kernels, so I did. So this does work.